src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs

```diff
@@ -8083,6 +8293,41 @@ public class Inner
             Diagnostic(ErrorCode.ERR_AttributeCantBeGeneric, "Inner").WithArguments("Outer<int>.Inner"));
         }
 
+        [Fact]
+        public void AliasedGenericAttributeType_NestedWithGenericAttributeFeature()
+        {
+            var source = @"
+using InnerAlias = Outer<int>.Inner;
+using OuterAlias = Outer<int>;
+
+[InnerAlias]
+class Test
+{
+    [OuterAlias.Inner]
+    static void Main()
+    {
+    }
+}
+
+public class Outer<T>
+{
+    // Not a subtype of Attribute, since that wouldn't compile.
+    public class Inner 
+    {
+    }
+}
+";
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                    // (5,2): error CS0616: 'Outer<int>.Inner' is not an attribute class
+                    // [InnerAlias]
+                    Diagnostic(ErrorCode.ERR_NotAnAttributeClass, "InnerAlias").WithArguments("Outer<int>.Inner").WithLocation(5, 2),
+                    // (8,17): error CS0616: 'Outer<int>.Inner' is not an attribute class
+                    //     [OuterAlias.Inner]
+                    Diagnostic(ErrorCode.ERR_NotAnAttributeClass, "Inner").WithArguments("Outer<int>.Inner").WithLocation(8, 17));
+        }
```

Comment:
>} [](start = 8, length = 1)

Some exploratory test ideas:
- a generic attribute: `class MyAttribute<T> { MyAttribute(T parameter) { ... } }` with missing type arguments `[MyAttribute(1)]` // should error, no type inference
- a generic attribute `class MyAttribute<T1, T2> { ... }` with wrong number of type arguments `[MyAttribute<int>]` or `[MyAttribute<int, string, byte>]`
- a generic attribute `class MyAttribute<t> { ... }` with a generic type argument `[MyAttribute<List<int>>]`

I'll update this list as I brainstorm more.

Reply:
Done

Reply:
Added a test for ascenario: a generic attribute: class MyAttribute<T> { MyAttribute(T parameter) { ... } } with type arguments the attribute constructor is not supported [MyAttribute<SomeType>(new SomeType())] // should error,  CS0181	Attribute constructor parameter 'parameter' has type 'SomeType', which is not a valid attribute parameter type

Reply:
Added a test for a scenario: a generic attribute class ```MyAttribute<T> { MyAttribute(T parameter) { ... } }``` and use it with a generic type parameter from a class ```[MyAttribute<U>]class Foo<U> {}``` should throw error CS0416: 'U': an attribute argument cannot use type parameters